From e3ab2a02e243eafe24586e34b4e3484525960b85 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <baimingcong@uniontech.com>
Date: Fri, 28 Jun 2024 11:05:23 +0800
Subject: [PATCH 072/360] mtd: nand: fix phytium-nand on 6.9

- Declare multiple functions as static functions to suppress warnings about
  missing prototypes.
- Drop exports for local GPL symbols.

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>
---
 drivers/mtd/nand/raw/phytium_nand.c | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/drivers/mtd/nand/raw/phytium_nand.c b/drivers/mtd/nand/raw/phytium_nand.c
index 263f1ba11999..e7cab4fe765e 100644
--- a/drivers/mtd/nand/raw/phytium_nand.c
+++ b/drivers/mtd/nand/raw/phytium_nand.c
@@ -236,7 +236,7 @@ static void phytium_nfc_parse_instructions(struct nand_chip *chip,
 	}
 }
 
-int phytium_nfc_prepare_cmd(struct nand_chip *chip,
+static int phytium_nfc_prepare_cmd(struct nand_chip *chip,
 			    struct phytium_nfc_op *nfc_op,
 			    enum dma_data_direction direction)
 {
@@ -254,9 +254,8 @@ int phytium_nfc_prepare_cmd(struct nand_chip *chip,
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(phytium_nfc_prepare_cmd);
 
-int phytium_nfc_send_cmd(struct nand_chip *chip,
+static int phytium_nfc_send_cmd(struct nand_chip *chip,
 			 struct phytium_nfc_op *nfc_op)
 {
 	struct phytium_nfc *nfc = to_phytium_nfc(chip->controller);
@@ -289,9 +288,8 @@ int phytium_nfc_send_cmd(struct nand_chip *chip,
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(phytium_nfc_send_cmd);
 
-int phytium_nfc_prepare_cmd2(struct nand_chip *chip,
+static int phytium_nfc_prepare_cmd2(struct nand_chip *chip,
 			     struct phytium_nfc_op *nfc_op,
 			     enum dma_data_direction direction,
 			     u32 cmd_num)
@@ -308,9 +306,8 @@ int phytium_nfc_prepare_cmd2(struct nand_chip *chip,
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(phytium_nfc_prepare_cmd2);
 
-int phytium_nfc_send_cmd2(struct nand_chip *chip,
+static int phytium_nfc_send_cmd2(struct nand_chip *chip,
 			  struct phytium_nfc_op *nfc_op,
 			  u32 cmd_num)
 {
@@ -350,9 +347,8 @@ int phytium_nfc_send_cmd2(struct nand_chip *chip,
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(phytium_nfc_send_cmd2);
 
-int phytium_nfc_wait_op(struct nand_chip *chip,
+static int phytium_nfc_wait_op(struct nand_chip *chip,
 			u32 timeout_ms)
 {
 	struct phytium_nfc *nfc = to_phytium_nfc(chip->controller);
@@ -382,7 +378,6 @@ int phytium_nfc_wait_op(struct nand_chip *chip,
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(phytium_nfc_wait_op);
 
 static int phytium_nfc_xfer_data_pio(struct phytium_nfc *nfc,
 				     const struct nand_subop *subop,
@@ -429,7 +424,7 @@ static int memcpy_to_reg16(struct phytium_nfc *nfc, u32 reg, u16 *buf, size_t le
 	return 0;
 }
 
-int phytium_nfc_default_data_interface(struct phytium_nfc *nfc)
+static int phytium_nfc_default_data_interface(struct phytium_nfc *nfc)
 {
 	int value;
 
@@ -493,7 +488,6 @@ int phytium_nfc_default_data_interface(struct phytium_nfc *nfc)
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(phytium_nfc_default_data_interface);
 
 static int phytium_nfc_naked_waitrdy_exec(struct nand_chip *chip,
 					  const struct nand_subop *subop)
@@ -1846,7 +1840,7 @@ static int phytium_nfc_init_dma(struct phytium_nfc *nfc)
 	return 0;
 }
 
-int phytium_nfc_init(struct phytium_nfc *nfc)
+static int phytium_nfc_init(struct phytium_nfc *nfc)
 {
 	u32 value;
 
@@ -1881,7 +1875,6 @@ int phytium_nfc_init(struct phytium_nfc *nfc)
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(phytium_nfc_init);
 
 static int phytium_nfc_setup_interface(struct nand_chip *chip, int chipnr,
 				       const struct nand_interface_config *conf)
-- 
2.20.1

